From f146d9a60f197fbf868be7eece68ff5cc58af4ff Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.org>
Date: Thu, 11 Jul 2019 15:12:05 +0100
Subject: [PATCH] drm/vc4: Pass the drm vrefresh to the firmware on
 mode set

More for completeness than need, but use drm_mode_vrefresh
to compute the vrefresh value, and pass that down to the
firmware on mode set.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.org>
---
 drivers/gpu/drm/vc4/vc4_firmware_kms.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

--- a/drivers/gpu/drm/vc4/vc4_firmware_kms.c
+++ b/drivers/gpu/drm/vc4/vc4_firmware_kms.c
@@ -768,8 +768,8 @@ static void vc4_crtc_mode_set_nofb(struc
 		      mode->hdisplay, mode->hsync_start, mode->hsync_end,
 		      mode->htotal, mode->hskew, mode->vdisplay,
 		      mode->vsync_start, mode->vsync_end, mode->vtotal,
-		      mode->vscan, mode->vrefresh, mode->picture_aspect_ratio,
-		      mode->flags);
+		      mode->vscan, drm_mode_vrefresh(mode),
+		      mode->picture_aspect_ratio, mode->flags);
 	mb.timings.display = vc4_crtc->display_number;
 
 	mb.timings.video_id_code = frame.avi.video_code;
@@ -785,7 +785,7 @@ static void vc4_crtc_mode_set_nofb(struc
 	mb.timings.vsync_end = mode->vsync_end;
 	mb.timings.vtotal = mode->vtotal;
 	mb.timings.vscan = mode->vscan;
-	mb.timings.vrefresh = 0;
+	mb.timings.vrefresh = drm_mode_vrefresh(mode);
 	mb.timings.flags = 0;
 	if (mode->flags & DRM_MODE_FLAG_PHSYNC)
 		mb.timings.flags |= TIMINGS_FLAGS_H_SYNC_POS;
